### ISSUE TITLE
dependency relation acl in French (and Spanish)

### ISSUE THREAD

**Issue by jheinecke** [UniversalDependencies/UD_French-GSD#4]
Hello

The UD annotation guidelines define the `acl `relation as 

> acl stands for finite and non-finite clauses that modify a nominal. The acl relation contrasts with the advcl relation, which is used for adverbial clauses that modify a predicate. The head of the acl relation is the noun that is modified, and the dependent is the head of the clause that modifies the noun. 

This implies the head of an `acl` deprel is only in rare cases a verb. However, in the UD_French-GSD (and UD_Spanish) I found 1542 `acl` (without `acl:relcl`) relations with a _verb_ as head  (24% of all `acl`, 2336 or 49% for UD_Spanish). All other treebanks (including French-sequoia, and Spanish-AnCora) have less than 1% of all `acl` relations linked to a verbal head. I wonder whether this is an error, since I expected `ccomp` (or `xcomp`) in these cases.

regards
Johannes

**Comment by dan-zeman** [comment]
I have checked the first ~ 25 occurrences in UD_Spanish-GSD and agree that they are either GSD annotation errors or errors in conversion from GSD to UD. Many of them should be `advcl`. Some seem to be examples of secondary predication and should be `xcomp`, or they should stay `acl` but be reattached to a nominal argument of the verb.

**Comment by dan-zeman** [comment]
Same situation in UD_Portuguese-GSD: 791 `acl` and 20 `acl:relcl`. A frequent (but not the only) example is a present participle _(gerundio)_ attached to a verb.

**Comment by dseddah** [comment]
Hi Dan an everyone,
Will the checking of that acl relation be included in a future version of the validator? 

Best,
Djamé

**Comment by dan-zeman** [comment]
@dseddah : I think it should. Sometimes it is difficult to establish that a rule must hold strictly in all languages but I think `acl` attached to a `VERB` might qualify as such a rule.

**Comment by bguil** [comment]
@jheinecke: I didn't manage to find the same numbers.
For the UD_Frensh-GSD corpora, I found:
 * in version 2.1: 447 cases of `acl` rel with a verbal head ([Grew-match](http://match.grew.fr/?custom=5b2d08505b64c&corpus=UD_French))
 * in version 2.2: 153 cases  ([Grew-match](http://match.grew.fr/?custom=5b2d09473d45b&corpus=UD_French-GSD@2.2))

How do you find the 1542 `acl` you mention?

**Comment by dan-zeman** [comment]
@bguil, @jheinecke : I also have 447 cases in UD_French-GSD 2.1: http://hdl.handle.net/11346/PMLTQ-XYAO

**Comment by dseddah** [comment]
maybe the next validator version could also include the notion of threshold below which a non-compliance to a rule should be qualified as residual errors, which are almost always unavoidable?

**Comment by dan-zeman** [comment]
If it is unavoidable because the grammar allows it, then it is not a matter for the validator.

If it is "unavoidable" because annotators make errors, then it is actually avoidable :-)

**Comment by dseddah** [comment]
After all this time treebanking, I don't believe in error-prone annotations for anything longer than a toy dataset  (I'm sure you don't either btw.)

**Comment by jheinecke** [comment]
@bguil , @dan-zeman You are right, I took by error the version 2.0. Much has been corrected since. After I `git pull` I only have 150 acl's linked to a verb (3%)

**Comment by dan-zeman** [comment]
@dseddah : You're right, I don't believe in error-free annotation. But I do believe that a subset of the errors can be detected automatically and eliminated (although the elimination sometimes has to be done manually).

**Comment by dseddah** [comment]
" I do believe that a subset of the errors can be detected automatically and eliminated although the elimination sometimes has to be done manually" 
Yes, but that's strictly a fonction of time and ressources available for that task, so when those are not available anymore, we need to learn how to live with those residual errors, like those 3% of POS annotations errors in the PTB, lying there for 25 year or so. We survived :)

**Comment by amir-zeldes** [comment]
Well, the beauty of releasing corpora on GitHub is that if we find those POS errors, instead of getting to know and learn them by heart for all those years, we can just fix them, commit, and they're gone in the next release :)

But on the current topic: I don't think we should forbid VERB --acl:relcl--> __ since sometimes this is a legitimate construction. For example in English:

http://match.grew.fr/?custom=5b2d3cd1a63d5&corpus=UD_English-GUM@2.2

**Comment by dan-zeman** [comment]
@amir-zeldes : I think there has been this discussion somewhere already but I cannot remember where. Anyways, [acl](http://universaldependencies.org/u/dep/acl.html) “stands for finite and non-finite clauses that modify a nominal”, which is not the case of the example you gave.

(BTW, some of the examples I saw in other languages resembled this English one. But it does not mean that their annotation is not a violation of the guidelines.)

**Comment by nschneid** [comment]
advcl:relcl, then?

**Comment by amir-zeldes** [comment]
@dan-zeman I didn't know that, so what is the current recommendation? I don't know whether a special subtype of relative clauses is needed for when they modify verbs (it would be quite rare), and if so, I think advcl:relcl is maybe the wrong way around: I think it's a relative clause first, as indicated by the relative pronoun, and happens to modify a verb second (so maybe acl:advcl?)

But if you think about it, relative clauses can modify all sorts of things and I'm not sure this is reason enough to give different labels:

NOUN (normal): the *day* which we agreed on
VERB: she *returned*, which I thought was great
ADV: she's singing *now*, which is too late
ADJ: The shirt is *green*, which fits me perfectly

The class of token being modified is already marked on that token's part of speech, so I don't see what is gained by more subtyping.

**Comment by dan-zeman** [comment]
I do not know what is the current recommendation (@sebschu, @manning?) But if anything with a relative pronoun qualifies as a relative clause, then it was probably wrong to assume that relative clauses are always a special case of `acl`.

@amir-zeldes : note that the part of speech of the parent token will not tell you everything. If it is a `NOUN`, you still don't know whether `acl` is correct:

1. _This is the car which I want to buy._ ... `acl`, modifies just the noun _car_.
2. _Ivan is a dancer, which is something I'll never understand._ ... modifies the clause, should be treated as if it modifies a verb.

**Comment by jnivre** [comment]
It is yet another case where the correlation between structural form (relative clause) and syntactic function (modifying a nominal) is less than perfect. We need to review all these cases systematically for v3, but for the time being I think the best compromise is to extend the use of acl:relcl to cover 2 as well (better than introducing advcl:relcl, since it is not really an adverbial clause either).

**Comment by amir-zeldes** [comment]
@dan-zeman yes, that's a good point, it's like nmod modifying a nominal predicate vs obl modifying the whole predication.

I'm with @jnivre in preferring acl:relcl for these and not modifying the current inventory if possible.

**Comment by nschneid** [comment]
Relative clauses seem like such a common strategy cross-linguistically that it feels odd to have them as a subtype anyway. Maybe we should make `relcl` a universal relation for v3.

**Comment by perrier54** [comment]
All the examples given by @amir-zeldes are translated in French with _"ce que"_:
_she's singing now, which is too late - elle chante maintenant, ce qui est trop tard_
The pronoun _"ce"_ is the antecedent of the relative pronoun and there is an anaphoric relation between _"elle chante maintenant"_ and _"ce",_ which is represented in the UD annotation with a PARATAXIS dependency from _"chante"_ to _"ce"_. 
In a similar way, it might be reasonable in English to put a PARATAXIS dependency from _"singing"_ to _"late"_;

**Comment by amir-zeldes** [comment]
For me parataxis, notwithstanding its use for parentheticals, implies a certain degree of independence - it's like a coordination without an explicit conjunction. These relative cases seem clearly subordinate for me, at least in English, so I wouldn't want to use that label. 

In French I guess you could say 'ce qui ...' is an independent NP, so it's more natural to use parataxis (there's no formal subordination, although both sides of the parataxis are not of the same category, which is maybe unusual).
